app/core/store/modules/teacherDashboard.js

```diff
@@ -304,6 +308,28 @@ export default {
       await Promise.all(fetchPromises)
     },
 
+    // Asssessments page
+    // options.data = { users: '', levelSessions: '' } -> properties needed for these objects, i.e. will be used as `project` in db queries
+    async fetchDataStudentAssessments ({ state, dispatch }, options = {}) {
+      const fetchPromises = []
+
+      fetchPromises.push(dispatch('courseInstances/fetchCourseInstancesForClassroom', state.classroomId, { root: true }))
+      fetchPromises.push(dispatch('courses/fetchReleased', undefined, { root: true }))
+      fetchPromises.push(dispatch('classrooms/fetchClassroomsForTeacher', { teacherId: state.teacherId }, { root: true }))
+      fetchPromises.push(dispatch('teacherDashboard/fetchClassroomData', options, { root: true }))
+      fetchPromises.push(dispatch('levels/fetchForClassroom', state.classroomId, { root: true }))
+
+      await Promise.all(fetchPromises)
+    },
+
+    // Assessments page - without blocking loading indicator
```

Comment:
I think it's because the Async function loads data that are not required for the initial rendering.
If you check above, the pattern is quite similar for other fetchX/fetchXAsync methods.